digi/xbee/reader.py

```diff
@@ -1054,46 +1056,76 @@ def __try_add_remote_device(self, xbee_packet):
 
-        'Special' means that this XBee has its API Output Mode distinct than Native (it's expecting
-        explicit data packets), but some device has sent it a non-explicit data packet (TransmitRequest f.e.).
+        This means that this XBee has its API Output Mode distinct than Native (it's expecting
+        explicit data packets), but some device has sent it a non-explicit data packet
+        (TransmitRequest f.e.).
         In this case, this XBee will receive a explicit data packet with the following values:
 
             1. Source endpoint = 0xE8
             2. Destination endpoint = 0xE8
             3. Cluster ID = 0x0011
             4. Profile ID = 0xC105
         """
-        if (xbee_packet.source_endpoint == 0xE8 and xbee_packet.dest_endpoint == 0xE8 and
-                xbee_packet.cluster_id == 0x0011 and xbee_packet.profile_id == 0xC105):
-            return True
-        return False
+        return (xbee_packet.source_endpoint == 0xE8 and xbee_packet.dest_endpoint == 0xE8
+                and xbee_packet.cluster_id == 0x0011 and xbee_packet.profile_id == 0xC105)
+
+    @staticmethod
+    def __is_explicit_io_packet(xbee_packet):
+        """
+        Checks if the provided explicit data packet is directed to the IO cluster.
+
+        This means that this XBee has its API Output Mode distinct than Native (it's expecting
+        explicit data packets), but some device has sent an IO sample packet
+        (IODataSampleRxIndicatorPacket f.e.).
+        In this case, this XBee will receive a explicit data packet with the following values:
+
+            1. Source endpoint = 0xE8
+            2. Destination endpoint = 0xE8
+            3. Cluster ID = 0x0092
+            4. Profile ID = 0xC105
+        """
+        return (xbee_packet.source_endpoint == 0xE8 and xbee_packet.dest_endpoint == 0xE8
+                and xbee_packet.cluster_id == 0x0092 and xbee_packet.profile_id == 0xC105)
 
     def __expl_to_no_expl(self, xbee_packet):
         """
         Creates a non-explicit data packet from the given explicit packet depending on
         this listener's XBee device protocol.
 
         Returns:
-            :class:`.XBeeAPIPacket`: the proper receive packet depending on the current protocol and the
-                available information (inside the packet).
+            :class:`.XBeeAPIPacket`: the proper receive packet depending on the current protocol and
+                the available information (inside the packet).
         """
         x64addr = xbee_packet.x64bit_source_addr
         x16addr = xbee_packet.x16bit_source_addr
         if self.__xbee_device.get_protocol() == XBeeProtocol.RAW_802_15_4:
             if x64addr != XBee64BitAddress.UNKNOWN_ADDRESS:
-                new_packet = RX64Packet(x64addr, 0, xbee_packet.receive_options, xbee_packet.rf_data)
+                new_pkt = RX64Packet(x64addr, 0, xbee_packet.receive_options,xbee_packet.rf_data)
```

Comment:
NP: missing whitespace before the last parameter of the packet constructor.

Reply:
Done.